src/Symfony/Bridge/ProxyManager/LazyProxy/PhpDumper/ProxyDumper.php

```diff
@@ -68,9 +68,9 @@ public function getProxyFactoryCode(Definition $definition, $id)
     {
         $instantiation = 'return';
 
-        if (ContainerInterface::SCOPE_CONTAINER === $definition->getScope()) {
+        if ($definition->isShared() && ContainerInterface::SCOPE_CONTAINER === $definition->getScope(false)) {
```

Comment:
We will need to bump the DependencyInjection component version in `composer.json` of the bridge or check if the method exists first.